### ISSUE TITLE
Issue with keeping OriginalCase

### ISSUE THREAD

**Issue by chrisjust** [zzzprojects/html-agility-pack#237]
Creating a HtmlNode
Changing the innerHtml on said node creates a childNode 
This childNode doesn't inherit from OwnerDocument's settings
#### To recreate: 
```csharp
var html = HtmlNode.CreateNode("<div class=\"table-responsive\"></div>");
// setting OwnerDocument's case
html.OwnerDocument.OptionOutputOriginalCase = true;
html.InnerHtml = "<SomeHtmlComponent></SomeHtmlComponent>";
Console.WriteLine(html.InnerHtml); 
// <somehtmlcomponent></somehtmlcomponent>
```

#### A workaround I've made is to set the setting on the ChildNode
For simplicity: 
```csharp
var html = HtmlNode.CreateNode("<div class=\"table-responsive\"></div>");
html.InnerHtml = "<SomeHtmlComponent></SomeHtmlComponent>";
html.ChildNodes[0].OwnerDocument.OptionOutputOriginalCase = true;
Console.WriteLine(html.InnerHtml);
//<SomeHtmlComponent></SomeHtmlComponent>
```

I haven't found anything in your documentation but it could be me doing something wrong of course :-)

**Comment by JonathanMagnan** [comment]
Hello @chrisjust ,

Thank you for reporting. We started to investigate this weird issue.

We can easily reproduce it with your code.

We will continue to check it during the week.

Best Regards,

Jonathan

---

**Performance Libraries**
`context.BulkInsert(list, options => options.BatchSize = 1000);  `
[Entity Framework Extensions](http://entityframework-extensions.net/) • [Entity Framework Classic](http://entityframework-classic.net/) • [Bulk Operations](http://bulk-operations.net/) •  [Dapper Plus](http://dapper-plus.net/) •  [LinqToSql Plus](http://linqtosql-plus.net/)

**Runtime Evaluation**
`Eval.Execute("x + y", new {x = 1, y = 2}); // return 3`
[C# Eval Function](http://eval-expression.net/) • [SQL Eval Function](http://eval-sql.net/)

**Comment by JonathanMagnan** [comment]
Hello @chrisjust ,

We investigated this issue and you are right.

The `InnerHtml` create a new document. So the `ownerDocument` from the new nodes is not the same as the `ownerDocument` from the outer node.

```csharp
HtmlDocument doc = new HtmlDocument();
doc.LoadHtml(value);
RemoveAllChildren();
AppendChildren(doc.DocumentNode.ChildNodes);
```

Currently, we don't think we will do anything. We started a fix however, we highly believe some side impact will occur with code currently in production.

```csharp
foreach (var node in doc.Nodesid.Values)
{
    node._ownerdocument = _ownerdocument;
    node.Attributes.ToList().ForEach(x => x._ownerdocument = _ownerdocument);
}
```

We will for sure look at it again but for now, we will not touch this issue.

Let me know if that explains correctly the current cause of the issue.

Best Regards,

Jonathan
